.travis.yml

```diff
@@ -142,12 +126,56 @@ matrix:
             - libboost-log1.55-dev
             - ninja-build
 
+
 before_script:
-  - ./install_dependencies.sh
-  - export PATH=$PATH:$PWD/odb/odb-2.4.0-x86_64-linux-gnu/bin/
-  - cmake -DBUILD_STATIC=1 -DCMAKE_CXX_COMPILER=${CXX_COMPILER} -DCMAKE_C_COMPILER=${C_COMPILER} -DCMAKE_CXX_FLAGS="-Iodb/libodb-2.4.0 -Iodb/libodb-sqlite-2.4.0" -DCMAKE_EXE_LINKER_FLAGS="-Wl,-rpath,odb/ -Lodb/"  -DODB_PATH=$PWD/odb/ -G Ninja
+  - echo "installing libodb"
+  - mkdir -p odb && cd odb
+  - wget http://codesynthesis.com/download/odb/2.4/libodb-2.4.0.tar.bz2 -O /tmp/libodb.tar.bz2
+  - tar jxvf /tmp/libodb.tar.bz2
+  - cd libodb-2.4.0 && ./configure && make
+  - cd ..
 
-script:
-  - ninja -j 1 -v
-  - ninja test
+  - echo "installing libodb-sqlite"
+  - wget http://codesynthesis.com/download/odb/2.4/libodb-sqlite-2.4.0.tar.bz2 -O /tmp/libodb-sqlite.tar.bz2
+  - tar jxvf /tmp/libodb-sqlite.tar.bz2
+  - cd libodb-sqlite-2.4.0 && ./configure --with-libodb=../libodb-2.4.0 && make
+  - cd ..
+
+  - pwd
+  - |
+    if [[ $TRAVIS_OS_NAME == "linux" ]]; then
+      cp libodb-2.4.0/odb/.libs/libodb.a .
+      cp libodb-sqlite-2.4.0/odb/sqlite/.libs/libodb-sqlite.a .
 
+      cp libodb-2.4.0/odb/.libs/libodb.so .
+      cp libodb-sqlite-2.4.0/odb/sqlite/.libs/libodb-sqlite.so .
+
+
+      wget http://codesynthesis.com/download/odb/2.4/odb-2.4.0-x86_64-linux-gnu.tar.bz2 -O /tmp/odb.tar.bz2
```

Comment:
All the code in `before_script` is the same except for the strings 'odb-2.4.0-x86_64-linux' and 'odb-2.4.0-i686-macosx'. Maybe you could provide a parameter to the `install_dependencies.sh` script with values 'linux' or 'macosx'?